### PR TITLE
Update dependencies for 5.8.0

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: qa/supply-chain/imports.lock [hunk 1]
```diff
@@ -406,11 +424,23 @@ criteria = "safe-to-deploy"
 version = "1.0.17"
 notes = "plenty of unsafe pointer and vec tricks, but in well-structured and commented code that appears to be correct"
 
+[[audits.bytecode-alliance.audits.sharded-slab]]
+who = "Pat Hickey <phickey@fastly.com>"
+criteria = "safe-to-deploy"
+version = "0.1.4"
+notes = "I always really enjoy reading eliza's code, she left perfect comments at every use of unsafe."
```
Comment (daira) [inline qa/supply-chain/imports.lock:473]: <3

File: Cargo.lock [hunk 2]
```diff
@@ -858,14 +867,14 @@ version = "0.5.5"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "5444c27eef6923071f7ebcc33e3444508466a76f7a2b93da00ed6e19f30c1ddb"
 dependencies = [
- "windows-sys",
+ "windows-sys 0.48.0",
```
Comment (daira) [inline Cargo.lock:870]: I note that the windows-sys crates are quite big (windows-sys itself is 2.5 MiB, and most of the import lib crates it depends on are around 800 KiB each). I think this duplication would probably go away if we updated all of our crates, but it's worth keeping an eye on.

Comment (str4d) [reply Cargo.lock:870]: I've updated all crates we depend on to their latest releases; we are blocked on de-duplication here by these dependencies having new versions released.

**Comment by str4d** [comment]
This is the Rust dependency graph as of this PR:
2023-12-18-zcashd-rust-dependency-graph

File: contrib/debian/copyright [hunk 3]
```diff
@@ -154,7 +154,7 @@ Copyright: 1994, 1995, 1996, 1999, 2000, 2001, 2002, 2004, 2005, 2006, 2007 Free
  2013 Ericsson AB
  2014 AppDynamics Inc.
  2015-2016 Brocade Communications Systems Inc.
-License: LGPL-3+ with ZeroMQ exception
+License: MPL-2.0
```
Comment (daira) [inline contrib/debian/copyright:157]: MPL-2.0 is a "[file-based copyleft](https://fossa.com/blog/open-source-software-licenses-101-mozilla-public-license-2-0/)" licence; that is, if you were to change the files that are part of ZeroMQ then you'd have to release them under the same terms, but there's no restriction on linking or combining with code under different licences. I guess that's fine, and is approximately what the exception was doing.

File: contrib/debian/copyright [hunk 4]
```diff
@@ -1232,50 +1232,6 @@ License: ISC
  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
  SOFTWARE.
 
-License: LGPL
```
Comment (daira) [inline contrib/debian/copyright:1235]: I checked that no other packages were using this.

**Review by daira — APPROVED** [review]
utACK

File: contrib/debian/copyright (hunk 3 above)
Comment (str4d) [reply contrib/debian/copyright:157]: I left the `depends/patches/zeromq/*` as covered by the same license as their code, so my patch to fix a warning in their code satisfies this requirement.

**Comment by str4d** [comment]
New failure in this PR: gtests failed on x86_64 macOS 11. This is only a tier 3 platform (so technically we don't require builds or tests to pass yet), but given that the test failures are in tests of the block and transaction verification code, I want to know what changed.

**Comment by str4d** [comment]
Tests pass on macOS as of f4ce9f6c3e85773d8a3d5a7d4f0c0e08ca6a6e19 (https://github.com/zcash/zcash/actions/runs/7264268215?pr=6799), so it wasn't the Rust updates. My guess is Boost, but it could also plausibly be CMake or `cxx`.

**Comment by str4d** [comment]
As I suspected, gtests pass on macOS with 34cbaa4c1a34a2a642eb1ec38b5237209036c493 but fail with 8669b5fc8fcddf6c6ee1ea0b227170419ee3d47c, so the problem is Boost. If we can't figure out the problem today, I'll postpone the Boost upgrade.

**Comment by str4d** [comment]
[FIrce-pushed](https://github.com/zcash/zcash/compare/201218fd05afad2400c8087c734cb776ffe656c1..daac43736d24b0fc64b0624adfb7c03373959012) to postpone Boost 1.84.0 update, as we haven't figured out why gtests fail on macOS.

**Review by daira — APPROVED** [review]
utACK

**Pull request merged**